### DISCUSSION TITLE
Responsive Layout Issue

### DISCUSSION THREAD

**Discussion by JCookDev** [tailwindlabs/tailwindcss#9743] [Help]
Hi, I'm having trouble understanding why my layout isn't displaying properly unless I Zoom out to 67% in my browser. Below is a link to the github issue I posted. Please help! Thanks!
https://github.com/adrianhajdin/project_music_player/issues/15#issue-1436373043

**Comment by wongjn** [comment]
It could be that the size of the viewport is too narrow to display how you would like it, hence why zooming out to 67% works. We don't have enough information at the moment, but perhaps reducing the sizing of elements could mean more of the elements in the grid can display side-by-side.

**Reply by JCookDev** [reply]
@wongjn Thanks for your response. I tried adjusting the sizes of the elements, but I didn't get the results I was looking for.  I followed an online tutorial  to build this app and I have no clue where to start making adjustments to yield my desired results. Below are screenshots of the code related to the elements in question. This is a React.JS app which uses Tailwind css for styling its components.

The following components appearances are too large and I'm hoping to make the layout appear the same for all screen sizes (except for on mobile of course).

These are the styles applied to the elements for the app overall. I thought Lines 16 and 17  was the issue here but I was wrong.
<img width="1113" alt="Screen Shot 2022-11-08 at 2 23 57 PM" src="https://user-images.githubusercontent.com/100865025/200656380-6a956e21-3eac-45df-8bf3-76f5e48d1ab1.png">


These are the styles for the SongCard component. 
<img width="300" alt="Screen Shot 2022-11-08 at 2 47 47 PM" src="https://user-images.githubusercontent.com/100865025/200660812-2d690177-ef32-4cf9-a925-885763ea6e11.png">

<img width="1274" alt="Screen Shot 2022-11-08 at 2 28 51 PM" src="https://user-images.githubusercontent.com/100865025/200657372-324e18d9-92f5-4141-9be8-26de3471afbc.png">

These are the styles for the Sidebar component.
<img width="170" alt="Screen Shot 2022-11-08 at 2 48 28 PM" src="https://user-images.githubusercontent.com/100865025/200660937-923a0734-45ed-41f4-b68e-fa84103b61f9.png">

<img width="787" alt="Screen Shot 2022-11-08 at 2 31 56 PM" src="https://user-images.githubusercontent.com/100865025/200657947-8c33bffc-0ef3-40dc-b683-4620e6d6a9d3.png">

These styles apply to the top charts section.
<img width="342" alt="Screen Shot 2022-11-08 at 2 48 55 PM" src="https://user-images.githubusercontent.com/100865025/200661030-dc2bce57-e713-462d-8234-d887244f066e.png">

<img width="1280" alt="Screen Shot 2022-11-08 at 2 37 32 PM" src="https://user-images.githubusercontent.com/100865025/200658938-ef13a54b-7ea2-41e5-8b44-24538f491bc2.png">

<img width="1280" alt="Screen Shot 2022-11-08 at 2 38 37 PM" src="https://user-images.githubusercontent.com/100865025/200659124-87a61a93-3416-4e73-a29d-96692dba24b5.png">

<img width="977" alt="Screen Shot 2022-11-08 at 2 39 16 PM" src="https://user-images.githubusercontent.com/100865025/200659287-6d4d9e26-a76d-4c4f-b63c-f943194c6205.png">

Currently the web page looks like this ( Unless I zoom out to 67% in my browser)
<img width="820" alt="Screen Shot 2022-11-08 at 2 42 48 PM" src="https://user-images.githubusercontent.com/100865025/200659908-6c5f89de-929e-4791-a073-c6fbbaf140ea.png">

Below is what the page looks like after zooming out to 67%.  I want the web page to look like this on desktop and pc screens. But I don't want users to have to adjust their screen size for this appearance. 
<img width="815" alt="Screen Shot 2022-11-08 at 2 40 47 PM" src="https://user-images.githubusercontent.com/100865025/200659533-701284c3-09c1-465c-844e-20f46a1ce4de.png">

**Reply by wongjn** [reply]
The troubles you are experiencing seem to be coming from many of the dimensions being set at "rigid" values. For example, in the `<SongCard>` component, you have a `w-[250px]` width, and in the `<Sidebar>` component, you again have a `w-[240px]` width class. You may want to look at these areas and see if you could perhaps use percentages, use flex layout or grid layout to essentially remove these rigid dimension size properties and let the general desired layout drive the dimension of the elements instead.

**Reply by JCookDev** [reply]
Ok, your response makes sense. However, I believe I misunderstood what you meant by use flex or grid layout. I was under the impression that I was using flex layout. I am using the flex properties in my component styles. Is this not the same thing as flex layout? Or are you referring to something different?

**Reply by wongjn** [reply]
Yes, you are using flex layout already. However, you could utilize it in other areas or perhaps adjust the layout logic slightly to better achieve your desired layout.

**Reply by JCookDev** [reply]
Thanks! I'll try that. I'll let you know how it goes

**Reply by JCookDev** [reply]
@wongjn I've tried lowering the element sizes but I'm not getting the desired results. I have a question about the "xl" breakpoint queries. I'm seeing this particular query in several components and i'm wondering if this has something to do with my layout issues. I read about breakpoint queries in the Tailwind css docs. According to the docs, a laptop screen is considered "lg", which is about 1024 px. I'm developing on a Macbook Pro laptop. Could the "xl" query possibly the reason why everything on the page appears to be zoomed in?

**Reply by wongjn** [reply]
It could be, you can compare your viewport size with the breakpoint sizes to see which breakpoints it matches.

**Reply by JCookDev** [reply]
How do I do that?

**Reply by wongjn** [reply]
If you are using the default `screens` configuration, you can look at the [Responsive Design](https://tailwindcss.com/docs/responsive-design) or [`screens`](https://tailwindcss.com/docs/screens) documentation for the default values. The breakpoints use `min-width`, so if your viewport width is larger that the breakpoint, it matches.
